scripts/run_backend_tests.py

```diff
@@ -403,7 +437,7 @@ def check_test_results(tasks, task_to_taskspec, generate_coverage_report):
                 coverage = task.task_results[0].get_report()[-2]
                 if (
                         spec.test_target not in coverage_exclusions
-                        and coverage != 100):
+                        and float(coverage) != 100.0):
```

Comment:
Why is this needed?

Reply:
But why this code even worked previously?

Reply:
https://github.com/oppia/oppia/blob/dba1c084b90a784343954ad811912b4f22c7c697/scripts/run_backend_tests.py#L232

It worked because over the above line used be this

https://github.com/oppia/oppia/blob/046f5fdecb36adb9613efea55cd042ccf8234263/scripts/run_backend_tests.py#L215

we were appending a number to the `messages` array whose type is `List[str]` previously. That is why it worked. Now since we are converting to a string. I changed it over here for the condition to work.

It didn't make sense to make messages `List[Union[str, int]]` just for the coverage result part.

Even in tests we are using the coverage number as a str.

https://github.com/oppia/oppia/blob/dba1c084b90a784343954ad811912b4f22c7c697/scripts/run_backend_tests_test.py#L435

Reply:
Thanks for the explanation. It makes sense, although an ideal solution would probably be to somehow to pass the float directly (by making the `messages` variable and putting some dict with the report and coverage), but I think it is pretty minor problem so lets keep it as is.